src/core_plugins/kibana/public/visualize/listing/visualize_listing_table.js

```diff
@@ -0,0 +1,253 @@
+        isSorted={this.state.sortedColumn === 'type'}
+        isSortAscending={this.sortableProperties.isAscendingByName('type')}
+      >
+        Type
+      </KuiTableHeaderCell>
+    ];
+  }
+
+  renderRowCells(item) {
+    return [
+      <KuiTableRowCell key="title">
+        <a className="kuiLink" href={this.getUrlForItem(item)}>
+          {item.title}
+        </a>
+      </KuiTableRowCell>,
+      <KuiTableRowCell key="type">
+        <span className="kuiStatusText">
+          {this.renderItemTypeIcon(item)}
+          {item.type.title}
+        </span>
+      </KuiTableRowCell>
+    ];
+  }
+
+  createRows() {
+    return this.state.pageOfItems.map(item => ({
+      id: item.id,
+      cells: this.renderRowCells(item)
+    }));
+  }
+
+  closeModal = () => {
+    this.setState({ showDeleteModal: false });
+  };
+
+  renderConfirmDeleteModal() {
+    return (
+      <KuiModalOverlay>
+        <KuiConfirmModal
+          message="Are you sure you want to delete the selected visualizations? This action is irreversible!"
+          title="Warning"
+          onCancel={this.closeModal}
+          onConfirm={this.deleteSelectedItems}
+          cancelButtonText="Cancel"
+          confirmButtonText="Delete"
+        />
+      </KuiModalOverlay>
+    );
+  }
+
+  onDelete = () => {
+    this.setState({ showDeleteModal: true });
+  };
+
+  deleteSelectedItems = () => {
+    this.props.deleteSelectedItems(this.state.selectedRowIds)
+      .then(() => this.fetchItems(this.state.filter))
+      .catch(() => {})
+      .then(() => this.deselectAll())
+      .then(() => this.closeModal());
+  };
+
+  onItemSelectionChanged = (newSelectedIds) => {
+    this.setState({ selectedRowIds: newSelectedIds });
+  };
+
+  onCreate() {
+    window.location = '#/visualize/new';
+  }
+
+  renderToolbarActions() {
+    return this.state.selectedRowIds.length > 0 ?
+      <KuiListingTableDeleteButton onDelete={this.onDelete} aria-label="Delete selected visualizations"/> :
+      <KuiListingTableCreateButton onCreate={this.onCreate} aria-label="Create new visualization"/>;
+  }
+
+  renderPager() {
+    return (
+      <KuiPager
+        startNumber={this.state.pageStartNumber}
+        hasNextPage={this.pager.hasNextPage}
+        hasPreviousPage={this.pager.hasPreviousPage}
+        endNumber={this.pager.endItem}
+        totalItems={this.items.length}
+        onNextPage={this.onNextPage}
+        onPreviousPage={this.onPreviousPage}
+      />
+    );
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.showDeleteModal && this.renderConfirmDeleteModal()}
+        <KuiListingTable
+          pager={this.renderPager()}
+          toolbarActions={this.renderToolbarActions()}
+          selectedRowIds={this.state.selectedRowIds}
+          rows={this.createRows()}
+          columns={this.renderColumns()}
+          onFilter={this.fetchItems}
+          filter={this.state.filter}
+          noItemsPrompt={<NoVisualizationsPrompt />}
```

Comment:
How about we combine `noItemsPrompt` and `loading` into a single `prompt` prop, and offload the logic for determining when to show which into the consumer? If a `prompt` is specified, it will be presented instead of the table. This gives the consumer some more control over exactly what the prompt looks like and what its content is.